src/gbinder_writer_p.h

```diff
@@ -165,6 +165,20 @@ gbinder_writer_data_append_local_object(
     GBinderLocalObject* obj)
     GBINDER_INTERNAL;
 
+#if GBINDER_FMQ_SUPPORTED
+
+void
+gbinder_writer_append_fmq_descriptor_hidl(
+    GBinderWriter* self,
+    const GBinderFmq* queue);
+
+void
+gbinder_writer_append_fmq_descriptor_aidl(
+    GBinderWriter* self,
+    const GBinderFmq* queue);
+
+#endif /* GBINDER_FMQ_SUPPORTED */
+
```

Comment:
how about moving these implementations inside `gbinder_rpc_protocol_hidl_write_fmq_descriptor` and `gbinder_rpc_protocol_aidl_write_fmq_descriptor`? that would significantly reduce the number of `#if GBINDER_FMQ_SUPPORTED` blocks and generally makes sense imo (considering that these functions are only called from those callbacks)

Reply:
Moved the implementations to gbinder_rpc_protocol.c.